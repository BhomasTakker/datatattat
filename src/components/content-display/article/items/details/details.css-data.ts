import { StackProps, StackTypeMap } from "@mui/material";
import { MARGINS } from "config/styles/styles.config";

export const spaceBetween: StackProps = {
	direction: "row",
	justifyContent: "space-between",
	marginTop: MARGINS.SMALL,
};

export const stack: StackProps = {
	direction: "column",
	marginTop: MARGINS.SMALL,
};

export const DetailsVariant = new Map([
	["space-between", spaceBetween],
	["stack", stack],
]);
