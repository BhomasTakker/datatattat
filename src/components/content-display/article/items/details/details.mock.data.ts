import { TIME_AGO_1_DAY, TIME_AGO_5_DAY } from "mockData/time";

// These need to be based off the same thing no?
// Else we will keep having t update etc
export const details1 = {
	docs: [],
	categories: ["category1", "category2"],
	authors: ["Author McAuther", "Authorson, Author"],
	publishers: ["Publisher.com"],
	published: TIME_AGO_5_DAY,
	modified: TIME_AGO_1_DAY,
};
