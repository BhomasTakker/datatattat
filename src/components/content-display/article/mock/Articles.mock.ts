import {
	TIME_AGO_1_HOUR,
	TIME_AGO_1_WEEK,
	TIME_AGO_5_DAY,
} from "mockData/time";

export const Article1 = {
	title: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_5_DAY,
		modified: "",
	},
};

export const Article2 = {
	title:
		"Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_1_HOUR,
		modified: "",
	},
};

export const Article3 = {
	title:
		"Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_1_WEEK,
		modified: "",
	},
};

export const Articles = {
	title: "Article List Title",
	src: "https://news.sky.com/",
	description: "Article List Title - Lorem Ipsum ",
	guid: "0001",
	variant: "collection",
	details: {},
	media: {},
	pagination: {
		results: 0,
	},
	// Duplicate keys need to remedy
	items: [
		Article1,
		Article2,
		Article3,
		Article1,
		Article2,
		Article3,
		Article1,
		Article2,
		Article3,
	],
};
