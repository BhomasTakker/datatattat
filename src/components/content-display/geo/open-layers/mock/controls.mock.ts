import { format } from "ol/coordinate";
import { CreateControl } from "../controls/open-layers.controls";
import { createOpenLayersView } from "../view/open-layers.view";
import { getCurrentProjection } from "../projections/open-layers.projections";
import { createLayer } from "../layers/open-layers.layers";
import { getTileLayerSource } from "../layers/sources/open-layers.sources";

const view = createOpenLayersView({ projection: getCurrentProjection() });
const exampleLayer = createLayer(
	"TileLayer",
	{},
	getTileLayerSource("OSM", {})
);

export const mockOSControls: CreateControl[] = [
	{
		id: "FullScreen",
	},
	{
		// We want to associate id with relevant control
		id: "ZoomSlider",
		options: {
			duration: 1000,
		},
	},
	{
		id: "ZoomToExtent",
	},
	{
		id: "Zoom",
		options: { duration: 1000, delta: 10 },
	},
	{
		id: "Rotate",
		options: { autoHide: false },
	},
	{
		id: "Attributions",
		options: { collapsible: false, collapsed: false },
	},
	{
		id: "ScaleLine",
	},
	{
		id: "Overview",
		options: {
			collapsible: false,
			layers: [exampleLayer],
			view,
		},
	},
	{
		id: "MousePosition",
		options: {
			projection: "EPSG:4326",
			coordinateFormat: (coordinate) => {
				if (!coordinate) return "";

				return format(coordinate, "{y}, {x}", 3);
			},
		},
	},
];
