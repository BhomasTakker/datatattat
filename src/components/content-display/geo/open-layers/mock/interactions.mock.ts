import DragBox, { DragBoxEvent } from "ol/interaction/DragBox";
import { setSelectInteraction } from "../interactions/select";
import { SelectEvent } from "ol/interaction/Select";
import KML from "ol/format/KML";
import GeoJSON from "ol/format/GeoJSON";
import { DragAndDropEvent } from "ol/interaction/DragAndDrop";
import { altKeyOnly, altShiftKeysOnly } from "ol/events/condition";
import { CreateInteraction } from "../hooks/useMapInteractions";

// Okay this shit can be done!
const dragBoxCb = (e: DragBoxEvent) => {
	console.log({ e });

	const dragBoxInteraction = e.target as DragBox;
	const extent = dragBoxInteraction.getGeometry().getExtent();
	// const extent = e.coordinate;

	// const interactions = e.mapBrowserEvent.map.getInteractions();

	console.log({ extent });
	e.mapBrowserEvent.map.getView().fit(extent);
	// const extent = dragBoxInteraction.getGeometry().getExtent();
	// map.getView().fit(extent as Extent);
};

const selectInteraction = setSelectInteraction({
	cb: (selectEvent) => console.log({ selectEvent }),
});

const formatConstructors = [KML, GeoJSON];

// mocking / selecting these might be difficult?
// if something uses select say
// we are going to need to create that interaction as part of what uses it
export const mockOLInteractions: CreateInteraction[] = [
	{
		id: "DragBox",
		options: { minArea: 64, cb: dragBoxCb, condition: altShiftKeysOnly },
	},
	{
		id: "Select",
		options: { cb: (selectEvent: SelectEvent) => console.log({ selectEvent }) },
	},
	// {
	// 	id: "Modify",
	// 	options: { cb: (modifyEvent: ModifyEvent) => console.log({ modifyEvent }) },
	// },
	{
		id: "DragAndDrop",
		options: {
			formatConstructors,
			cb: (dragAndDropEvent: DragAndDropEvent) =>
				console.log({ dragAndDropEvent }),
		},
	},
];
