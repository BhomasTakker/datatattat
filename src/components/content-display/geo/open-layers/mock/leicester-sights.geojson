{
	"type": "FeatureCollection",
	"features": [
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [
					[
						[-1.1288642441902823, 52.66407147616076],
						[-1.1509241628855875, 52.66738934138226],
						[-1.1770002296757411, 52.65257550397584],
						[-1.189303725978192, 52.633633213459575],
						[-1.1887528231581541, 52.622598334817326],
						[-1.1876510263188038, 52.61951223479471],
						[-1.1904055345964935, 52.614599238597265],
						[-1.1843456035817894, 52.59508276744009],
						[-1.165064004899449, 52.5965328614667],
						[-1.1272353446268824, 52.59675594859587],
						[-1.0938139069099009, 52.606459139265354],
						[-1.093630272636858, 52.61649466094761],
						[-1.0758177481396842, 52.65572261534621],
						[-1.1079537459436324, 52.66764005144748],
						[-1.1288642441902823, 52.66407147616076]
					]
				],
				"type": "Polygon"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1203762897064848, 52.62823007295199],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1111324536256575, 52.62663336950814],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1142300616747036, 52.62887172680553],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.126802779062416, 52.62826984161606],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.122353056064128, 52.627833204789965],
				"type": "Point"
			}
		}
	]
}
