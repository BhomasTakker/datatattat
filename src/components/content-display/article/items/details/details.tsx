import { Stack, Typography } from "@mui/material";
import { MARGINS } from "config/styles/styles.config";
import { Time } from "../time/Time";
import { Details } from "@/src/types/data-structures/base";
import { DetailsVariant } from "./details.css-data";
import { CSS } from "@/src/css/text";

interface DetailsTextProps {
	children: React.ReactNode;
}

const DetailsText = ({ children }: DetailsTextProps) => {
	return (
		<Typography
			sx={{
				...CSS.maxLines({ maxLines: 1 }),
			}}
		>
			{children}
		</Typography>
	);
};

export type DetailsVariantType = "space-between" | "stack";

export interface DetailsComponentProps {
	details?: Details;
	showPublished?: boolean;
	showCategories?: boolean;
	showpublishers?: boolean;
	showAuthors?: boolean;
	variant?: DetailsVariantType;
	// divider?: boolean;
}

export const DetailsComponent = ({
	details,
	showPublished = false,
	showCategories = false,
	showpublishers = false,
	showAuthors = false,
	// divider = true,
	variant = "space-between",
}: DetailsComponentProps) => {
	const {
		published = null,
		categories = null,
		publishers = null,
		authors = null,
	} = details || {};

	const variantProps = DetailsVariant.get(variant);

	return (
		<Stack
			data-testid="details"
			{...variantProps}
			// divider={React Element}
		>
			{/* do something proper with the time!! */}
			{/* Like we want an image AND avatar component - we want Time - variant age/publish date */}
			{showPublished && <Time time={published} variant="age" />}
			{showCategories && <DetailsText>{categories?.join(", ")}</DetailsText>}
			{showpublishers && <DetailsText>{publishers?.join(", ")}</DetailsText>}
			{showAuthors && <DetailsText>{authors?.join(", ")}</DetailsText>}
		</Stack>
	);
};
