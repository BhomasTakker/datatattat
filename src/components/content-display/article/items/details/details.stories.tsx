import type { Meta, StoryObj } from "@storybook/react";

import { DetailsComponent } from "./details";
import { details1 } from "./details.mock.data";

const meta: Meta<typeof DetailsComponent> = {
	component: DetailsComponent,
};
export default meta;
type Story = StoryObj<typeof DetailsComponent>;

export const Standard: Story = {
	args: {
		details: details1,
		variant: "space-between",
		showPublished: false,
		showCategories: false,
		showpublishers: false,
		showAuthors: false,
	},
};
