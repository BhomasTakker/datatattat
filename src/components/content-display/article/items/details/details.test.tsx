import { render, screen } from "@testing-library/react";
import { DetailsComponent } from "./details";
import { details1 } from "./details.mock.data";

describe("Details", () => {
	it("should render empty Details", () => {
		render(<DetailsComponent />);
		const detailsElement = screen.getByTestId("details");
		expect(detailsElement).toBeInTheDocument();
	});

	it("should render Details with published data", () => {
		render(<DetailsComponent details={details1} showPublished />);
		// should - mock Time and return whatever
		const detailsElement = screen.getByText(/published/i);
		expect(detailsElement).toBeInTheDocument();
	});

	it("should render Details with categories data", () => {
		render(<DetailsComponent details={details1} showCategories />);
		const detailsElement = screen.getByText(/category1, category2/i);
		expect(detailsElement).toBeInTheDocument();
	});

	it("should render Details with publishers data", () => {
		render(<DetailsComponent details={details1} showpublishers />);
		const detailsElement = screen.getByText("Publisher.com");
		expect(detailsElement).toBeInTheDocument();
	});

	it("should render Details with authors data", () => {
		render(<DetailsComponent details={details1} showAuthors />);
		const detailsElement = screen.getByText(
			"Author McAuther, Authorson, Author"
		);
		expect(detailsElement).toBeInTheDocument();
	});

	describe("snapshots", () => {
		it("should match snapshot for space-between variant", () => {
			render(
				<DetailsComponent
					details={details1}
					variant="space-between"
					showPublished
					showCategories
					showpublishers
					showAuthors
				/>
			);
			const detailsElement = screen.getByTestId("details");
			expect(detailsElement).toMatchSnapshot();
		});

		it("should match snapshot for stack variant", () => {
			render(
				<DetailsComponent
					details={details1}
					variant="stack"
					showPublished
					showCategories
					showpublishers
					showAuthors
				/>
			);
			const detailsElement = screen.getByTestId("details");
			expect(detailsElement).toMatchSnapshot();
		});
	});
});
