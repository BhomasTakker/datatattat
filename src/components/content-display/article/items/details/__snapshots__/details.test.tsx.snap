// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details snapshots should match snapshot for space-between variant 1`] = `
<div
  class="css-1jligzp-MuiStack-root"
  data-testid="details"
>
  <p
    class="MuiTypography-root MuiTypography-body2 css-sbmvo9-MuiTypography-root"
    data-testid="time"
  >
    Published 5 days ago
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    category1, category2
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    Publisher.com
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    Author McAuther, Authorson, Author
  </p>
</div>
`;

exports[`Details snapshots should match snapshot for stack variant 1`] = `
<div
  class="css-1ds6d3x-MuiStack-root"
  data-testid="details"
>
  <p
    class="MuiTypography-root MuiTypography-body2 css-sbmvo9-MuiTypography-root"
    data-testid="time"
  >
    Published 5 days ago
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    category1, category2
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    Publisher.com
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
  >
    Author McAuther, Authorson, Author
  </p>
</div>
`;
